### PR TITLE
Try last version 1.14.1

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

**Comment by alkino** [comment]
The bug seems to be https://github.com/HDFGroup/hdf5/issues/1632

File: tests/unit/tests_high_five_base.cpp [hunk 1]
```diff
@@ -291,7 +291,7 @@ TEST_CASE("Test group properties") {
     File file(file_name, File::Truncate, fapl);
 
     GroupCreateProps props;
-    props.add(EstimatedLinkInfo(1000, 500));
+    props.add(EstimatedLinkInfo(10, 60));
```
Comment (matz-e) [inline tests/unit/tests_high_five_base.cpp:294]: We're giving up on this?

Comment (1uc) [reply tests/unit/tests_high_five_base.cpp:294]: I reread the all the messages. One of the very helpful things @matz-e wrote was that this is in debug mode. If we look closely what fails is an `HDassert`. However, it turns out that this is just a debugging help, i.e.,
https://github.com/HDFGroup/hdf5/blob/577c192518598c7e2945683655feffcdbdf5a91b/src/H5private.h#L601

I've updated the CI to build Release versions of HDF5; and the issue went away.

**Review by 1uc — APPROVED** [review]
Thank you!

File: tests/unit/tests_high_five_base.cpp (hunk 1 above)
Comment (matz-e) [reply tests/unit/tests_high_five_base.cpp:294]: After some deliberation, I think as long as upstream is dragging their feet, we should insist on this. So I'm fine with the change, luckily we don't depend on this feature 😅

**Review by matz-e — APPROVED** [review]

**Pull request merged**